pysmiles/read_smiles.py

```diff
@@ -193,8 +195,21 @@ def read_smiles(smiles, explicit_hydrogen=False, zero_order_bonds=True,
     else:
         mark_aromatic_edges(mol)
 
+    # This is a bit of an overreach, we should only add implicit hydrogens to
+    # atoms in the organic subset. However, all non-organic atoms already have
+    # a hcount, so all is well.
     fill_valence(mol)
 
+    if strict:
+        for node in mol:
+            element = mol.nodes[node].get('element', '*')
+            if element != '*' and element not in PTE:
+                raise KeyError(f'Unknown element {element}')
+            elif element != '*' and bonds_missing(mol, node):
```

Comment:
This happens when say the bond orders don't match up with the elements default valency?

Reply:
Yes. If we can make a reasonable guess how many bonds an element should have, and it has a different amount, we complain.